src/token_cache.rs

```diff
@@ -132,34 +149,33 @@ impl TokenCache {
     ) -> Option<RegistryTokenType> {
         let registry = reference.resolve_registry().to_string();
         let repository = reference.repository().to_string();
-        match self
-            .tokens
-            .read()
-            .await
-            .get(&(registry.clone(), repository.clone(), op))
-        {
-            Some((ref token, expiration)) => {
+        let key = TokenCacheKey {
+            registry,
+            repository,
+            operation: op,
+        };
+        match self.tokens.read().await.get(&key) {
+            Some(TokenCacheValue {
+                ref token,
+                expiration,
+            }) => {
                 let now = SystemTime::now();
                 let epoch = now
                     .duration_since(UNIX_EPOCH)
                     .expect("Time went backwards")
                     .as_secs();
                 if epoch > *expiration {
-                    debug!(%registry, %repository, ?op, %expiration, miss=false, expired=true, "Fetching token");
+                    debug!(?key, %expiration, miss=false, expired=true, "Fetching token");
```

Comment:
I think we should still log each of the different fields separately here. The default debug implementation might be really "noisy" when trying to look at these logs and it will be hard to filter them based on operation or repo in any of the logging tools

Reply:
fixed